zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/usermanagement/RoleControllerTest.java

```diff
@@ -158,28 +161,63 @@ void updateRoleWithEmptyNameShouldFail() {
   @Test
   void updateNonExistingRoleShouldReturnError() {
     // given
-    final var roleKey = 100L;
+    final var roleId = "roleId";
     final var roleName = "Updated Role Name";
-    final var path = "%s/%s".formatted(ROLE_BASE_URL, roleKey);
-    when(roleServices.updateRole(roleKey, roleName))
+    final var description = "Updated Role Description";
+    final var request = new UpdateRoleRequest(roleId, roleName, description);
+    final var path = "%s/%s".formatted(ROLE_BASE_URL, roleId);
+    when(roleServices.updateRole(request))
         .thenReturn(
             CompletableFuture.failedFuture(
                 new CamundaBrokerException(
                     new BrokerRejection(
-                        RoleIntent.UPDATE, roleKey, RejectionType.NOT_FOUND, "Role not found"))));
+                        RoleIntent.UPDATE, 1L, RejectionType.NOT_FOUND, "Role not found"))));
 
     // when / then
     webClient
-        .patch()
+        .put()
         .uri(path)
         .accept(MediaType.APPLICATION_JSON)
         .contentType(MediaType.APPLICATION_JSON)
-        .bodyValue(new RoleUpdateRequest().changeset(new RoleChangeset().name(roleName)))
+        .bodyValue(new RoleUpdateRequest().name(roleName).description(description))
         .exchange()
         .expectStatus()
         .isNotFound();
 
-    verify(roleServices, times(1)).updateRole(roleKey, roleName);
+    verify(roleServices, times(1)).updateRole(request);
+  }
+
+  @Test
+  void updateRoleWithoutDescriptionShouldFail() {
+    // given
+    final var roleId = "roleId";
+    final var roleName = "roleName";
+    final String description = null;
```

Comment:
🔧 You can delete this and simply not provide the description in the body value

Reply:
I know, I liked this for clarification purposes more.